apis/v1alpha1/route_extension_reference_types.go

```diff
@@ -0,0 +1,34 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+// RouteMatchExtensionObjectReference identifies a route-match extension object
+// within a known namespace.
+//
+// +k8s:deepcopy-gen=false
+type RouteMatchExtensionObjectReference = ConfigMapsDefaultLocalObjectReference
```

Comment:
I do wonder if we should do 

`type RouteMatchExtensionObjectReference ConfigMapsDefaultLocalObjectReference` (no equals)

and it will fix all of our codegen issues.